test/test_server_selection.py

```diff
@@ -32,5 +36,74 @@
 class TestAllScenarios(create_selection_tests(_TEST_PATH)):
     pass
 
+
+class TestCustomServerSelectorFunction(IntegrationTest):
+    def test_functional_select_max_port_number_host(self):
+        # Selector that returns server with highest port number.
+        def custom_selector(selection):
```

Comment:
The spec says this should input and output a list of server descriptions:
> The function takes as a parameter a list of server descriptions representing the suitable servers for the read or write operation, and returns a list of server descriptions that should still be considered suitable.

We need to follow that part of the spec.